shared/src/lib.rs

```diff
@@ -127,37 +100,76 @@ pub enum EDHOCError {
     UnknownError = 9,
 }
 
-#[repr(C)]
 #[derive(Debug)]
-pub struct State<Phase: EDHOCState> {
-    pub current_state: PhantomData<Phase>,
-    pub x_or_y: BytesP256ElemLen,   // ephemeral private key of myself
-    pub c_i: u8,                    // connection identifier chosen by the initiator
-    pub gy_or_gx: BytesP256ElemLen, // g_y or g_x, ephemeral public key of the peer
+pub struct Start;
```

Comment:
How long does generating a key take? (I have no clue, can be just-take-the-few-bytes-of-entropy, can be do-some-complex-GF2-computations). If it's not trivial, it may be a good idea to store the x_or_y already in the Start, as that allows a system to prepare a key while not in use.

(Could also spin that out to an own low-prio issue)

Reply:
I only have data for preparing message 1, which as a whole takes about 20 ms, given these conditions: crypto initialisation is stateful (see #94); using the cc310 backend. So it is fast, although software backends might take longer.